packages/amplify-provider-awscloudformation/src/graphql-transformer/api-utils.ts

```diff
@@ -8,13 +8,13 @@ export async function searchablePushChecks(context, map, apiName): Promise<void>
     if (searchableModelTypes.length) {
       const currEnv = context.amplify.getEnvInfo().envName;
       const teamProviderInfo = stateManager.getTeamProviderInfo();
+      const version = await getTransformerVersion(context);
       const instanceType = _.get(
         teamProviderInfo,
-        [currEnv, 'categories', 'api', apiName, ResourceConstants.PARAMETERS.ElasticsearchInstanceType],
+        [currEnv, 'categories', 'api', apiName, version === 2 ? ResourceConstants.PARAMETERS.OpenSearchInstanceType : ResourceConstants.PARAMETERS.ElasticsearchInstanceType],
```

Comment:
Is it possible to model this as a fallthrough condition instead?

```ts
const getInstanceType(instanceTypeParam: string): string? => _.get(teamProviderInfo, [currEnv, 'categories', 'api', apiName, instanceTypeParam]);
const instanceType =
    getInstanceType(ResourceConstants.PARAMETERS.OpenSearchInstanceType) ??
    getInstanceType(ResourceConstants.PARAMETERS.ElasticsearchInstanceType) ??
    't2.small.elasticsearch';
```

I'd like to avoid explicitly checking for transformer version as much as possible.